pkg/sql/catalog/descs/collection.go

```diff
@@ -281,6 +362,119 @@ func (tc *Collection) WriteDesc(
 	return txn.Run(ctx, b)
 }
 
+// LookupDatabaseID returns the descriptor ID assigned to a database name.
+func (tc *Collection) LookupDatabaseID(
+	ctx context.Context, txn *kv.Txn, dbName string,
+) (descpb.ID, error) {
+	// First look up in-memory descriptors in collection,
+	// except for leased descriptors.
+	dbInMemory, err := func() (catalog.Descriptor, error) {
+		flags := tree.CommonLookupFlags{
+			Required:       true,
+			AvoidLeased:    true,
+			AvoidStorage:   true,
+			IncludeOffline: true,
+		}
+		return tc.getDescriptorByName(
+			ctx, txn, nil /* db */, nil /* sc */, dbName, flags, catalog.Database,
+		)
+	}()
+	if errors.IsAny(err, catalog.ErrDescriptorNotFound, catalog.ErrDescriptorDropped) {
+		// Swallow these errors to fall back to storage lookup.
+		err = nil
+	}
+	if err != nil {
+		return descpb.InvalidID, err
+	}
+	if dbInMemory != nil {
+		return dbInMemory.GetID(), nil
+	}
+	// Look up database ID in storage if nothing was found in memory.
+	return tc.stored.LookupDescriptorID(ctx, txn, keys.RootNamespaceID, keys.RootNamespaceID, dbName)
```

Comment:
I don't understand why we need this method, I think `tc.getDescriptorByName` already look through stored layer and, before that, we resolve name from `uncommitted layer`, no?

Same question for the two new methods below.

Reply:
No, in that `getDescriptorByName ` call the stored layer was explicitly bypassed by setting the `AvoidStorage` flag to true.

Reply:
Nice. my bad missing that flag..